velodyne_driver/include/velodyne_driver/time_conversion.h

```diff
@@ -0,0 +1,46 @@
+#ifndef VELODYNE_DRIVER_TIME_CONVERSION_HPP
+#define VELODYNE_DRIVER_TIME_CONVERSION_HPP
+
+#include <ros/ros.h>
+#include <ros/time.h>
+
+/** @brief Function used to check that hour assigned to timestamp in conversion is
+ * correct. Velodyne only returns time since the top of the hour, so if the computer clock
+ * and the velodyne clock (gps-synchronized) are a little off, there is a chance the wrong
+ * hour may be associated with the timestamp
+ * 
+ * @param stamp timestamp recovered from velodyne
+ * @param nominal_stamp time coming from computer's clock
+ * @return timestamp from velodyne, possibly shifted by 1 hour if the function arguments
```

Comment:
Should we throw a warning if this is happening

Reply:
An error will happen in `TopicDiagnostic` via the `TimeStampStatusParam` bounds